gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/avatarpicker/AvatarPicker.kt

```diff
@@ -135,6 +144,38 @@ internal fun AvatarPicker(
 internal fun AvatarPicker(uiState: AvatarPickerUiState, onEvent: (AvatarPickerEvent) -> Unit) {
     val context = LocalContext.current
     var loadingSectionHeight by remember { mutableStateOf(DEFAULT_PAGE_HEIGHT) }
+    var storagePermissionRationaleDialogVisible by rememberSaveable { mutableStateOf(false) }
+    var avatarToDownload: Avatar? by remember { mutableStateOf(null) }
+
+    val writeExternalStoragePermissionLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.RequestPermission(),
+    ) { isGranted: Boolean ->
+        if (isGranted) {
+            avatarToDownload?.let { onEvent(AvatarPickerEvent.DownloadAvatarTapped(it)) }
+            avatarToDownload = null
+        } else {
+            storagePermissionRationaleDialogVisible = true
+        }
+    }
```

Comment:
❓ Shouldn't we set to `null` the `avatarToDownload` in both cases?

```suggestion
        if (isGranted) {
            avatarToDownload?.let { onEvent(AvatarPickerEvent.DownloadAvatarTapped(it)) }
        } else {
            storagePermissionRationaleDialogVisible = true
        }
        avatarToDownload = null
    }
```